File: json/json-annotation-processor/src/test/java/ru/tinkoff/kora/json/annotation/processor/dto/DtoWithInnerDto.java
package ru.tinkoff.kora.json.annotation.processor.dto;

import ru.tinkoff.kora.json.common.annotation.Json;

import java.util.List;
import java.util.Map;

@Json
public record DtoWithInnerDto(
    InnerDto inner,
    List<InnerDto> field2,
    Map<String, InnerDto> field3,
    List<List<InnerDto>> field4) {

    @Json
    public record InnerDto(String field1) {

    }
}

File: json/json-symbol-processor/src/test/kotlin/ru/tinkoff/kora/json/ksp/dto/DtoWithInnerDto.kt
package ru.tinkoff.kora.json.ksp.dto

import ru.tinkoff.kora.json.common.annotation.Json

@Json
data class DtoWithInnerDto(
    val inner: InnerDto?,
    val field2: List<InnerDto>,
    val field3: Map<String, InnerDto>,
    val field4: List<List<InnerDto>>
) {
    @Json
    data class InnerDto(val field1: String)
}
